# **Problem statement:** 
A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.
   Minitab File : Cutlets.mtw



We set the hypothesis as

Ho:There is a no significance difference between the diameter of the cutlet.

H1:There is a significant difference between the diameter of the cutlet.

In [3]:
import pandas as pd
import numpy as np
from scipy import stats
data=pd.read_csv("/content/Cutlets.csv")
data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

Here, x is discrete and y is continous random variable . To check whether the variables y1 and y2 follow normal distribution , we use Shapiro test.

We set a hypothesis as 

H0:y1 and y2 are normal.

H1:y1 and y2 are not normal.

In [4]:
stats.shapiro(data["Unit A"])

(0.9649458527565002, 0.3199819028377533)

In [5]:
stats.shapiro(data["Unit B"])

(0.9727300405502319, 0.5224985480308533)

For unit A, p value =0.32 > alpha=0.05 

For unit B, p value =0.52 > alpha=0.05 

There , Ho is accepted at 5% l.o.s ,hence, y1 and y2 are normal.

In [7]:
#paired t test
stats.ttest_rel(data["Unit A"],data["Unit B"])

Ttest_relResult(statistic=0.7536787225614323, pvalue=0.45623007680384076)

pvalue=0.75 > alpha=0.05 

Therefore, H0 is accepted and we can conclude that there is no significant difference between the diameter of the cutlet.

#Problem Statement

 A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.
Minitab File: LabTAT.mtw



In [8]:
data1=pd.read_csv("/content/LabTAT.csv")
data1.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

We set the hypothesis as

Ho:There is a no significance difference between the average Turn Around Time (TAT)

H1:There is a significant difference between the average Turn Around Time (TAT)

# Normality test

To check whether the variables y1 ,y2 ,y3 and y4 follow normal distribution , we use Shapiro test.

We set a hypothesis as

H0:y1,y2,y3 and y4 are normal.

H1:y1,y2,y3 and y4 are not normal.

In [9]:
stats.shapiro(data1["Laboratory 1"])

(0.9901824593544006, 0.5506953597068787)

In [10]:
stats.shapiro(data1["Laboratory 2"])

(0.9936322569847107, 0.8637524843215942)

In [11]:
stats.shapiro(data1["Laboratory 3"])

(0.9886345267295837, 0.4205053448677063)

In [12]:
stats.shapiro(data1["Laboratory 4"])

(0.9913753271102905, 0.6618951559066772)

For Laboratory 1, p value =0.55 > alpha=0.05

For Laboratory 2, p value =0.86 > alpha=0.05

For Laboratory 3, p value =0.42 > alpha=0.05

For Laboratory 4, p value =0.66 > alpha=0.05

Therefore , Ho is accepted at 5% l.o.s ,hence, y1 ,y2,y3 and y4 are normal.

# Variance test

H0 :Variance of all 4 laboratories are equal.

H1 :Variance of all 4 laboratories are unequal.

In [13]:
stats.levene(data1["Laboratory 1"], data1["Laboratory 2"], data1["Laboratory 3"], data1["Laboratory 4"])

LeveneResult(statistic=2.599642500418024, pvalue=0.05161343808309816)

pvalue=0.51 >alpha=0.05 ,therefore we can say that variance of all 4 laboratories are somewhat equal.

#One-way Anova

In [14]:
stats.f_oneway(data1["Laboratory 1"], data1["Laboratory 2"], data1["Laboratory 3"], data1["Laboratory 4"])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

Since, pvalue is less than alpha , we reject H0 at 5%l.o.s,and we conclude that there is a significant difference between the average Turn Around Time (TAT)



#Problem Statement

Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.
Buyer Ratio.mtw



In [4]:
data2=pd.read_csv("/content/BuyerRatio.csv")
data2.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

We set the hypothesis as 

H0:male-female buyer rations are similar across regions.

H1:male-female buyer rations are not similar across regions.

In [5]:
stats.chi2_contingency([data2["East"], data2["West"], data2["North"], data2["South"]])

(1.5959455386610577,
 0.6603094907091882,
 3,
 array([[  42.76531299,  442.23468701],
        [ 146.81287862, 1518.18712138],
        [ 131.11756787, 1355.88243213],
        [  72.30424052,  747.69575948]]))

Since, pvalue=0.66 > alpha=0.5 ,we accetep H0 at 5% l.o.s .
Therefore,we can say that male-female buyer rations are similar across regions.

# Problem Statement
TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences
Minitab File: CustomerOrderForm.mtw


In [7]:
data3 = pd.read_csv("/content/Costomer+OrderForm.csv")
data3.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

We set the hypothesis as

H0:The defective % does not varies by centre

H1:The defective % does  varies by centre

In [8]:
stats.chi2_contingency([data3['Phillippines'].value_counts(), data3['Indonesia'].value_counts(), data3['Malta'].value_counts(), data3['India'].value_counts()])

(3.8589606858203545, 0.2771020991233144, 3, array([[271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25]]))

Since, pvalue=0.277 > alpha=0.5 ,we accetep H0 at 5% l.o.s .The defective % does not varies by centre 